projboard/tests/test_article.py

```diff
@@ -0,0 +1,140 @@
+        sup = Subject(name='N')
+        sup1 = Subject(name='R')
+        sup.save()
+        sup1.save()
+        return sup, sup1
+
+    @pytest.fixture
+    @pytest.mark.django_db
+    def create_user(self):
+        user = User(name="Anas", nickname="A99", email="Anas@gmail.com", password="Anas123")
+        user1 = User(name="John", nickname="J99", email="John@gmail.com", password="John123")
+        user.save()
+        user1.save()
+        return user, user1
+
+    @pytest.fixture
+    @pytest.mark.django_db
+    def create_articles(self, create_user, create_supjects):
+        article = Article(user_id=create_user[0], title=TITLE, content=CONTENT, subject_id=create_supjects[0])
+        article1 = Article(user_id=create_user[1], title="RED HAT", content=CONTENT, subject_id=create_supjects[1])
+        article.save()
+        article1.save()
+        return article, article1
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_objectCreation(self, create_articles):
+        article = create_articles[0]
+        assert article.title == TITLE
+        assert article.content == CONTENT
+        # testing __str__ function
+        assert article.__str__() == article.title
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_PartFunctions(self, create_articles):
+        article = create_articles[0]
+        # testing @property functions => PartOf..()
+        # whent title & content length is less than reality
+        assert article.PartOfTitle(5) == article.title[:5] + '...'
+        assert article.PartOfContent(5) == article.content[:5] + '...'
+        # whent title & content length is greater than reality
+        assert article.PartOfTitle(100) == article.title[:100]
+        assert article.PartOfContent(100) == article.content[:100]
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_objectSavedInDb(self, create_articles):
+        assert create_articles[0] in Article.objects.all()
+        assert create_articles[1] in Article.objects.all()
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    # using Edite function
+    def test_EditeObject(self, create_articles, create_supjects):
+
+        article = create_articles[0]
+        title = "New Title"
+        content = "New Content"
+        subject = create_supjects[1]
+        old_subject = create_supjects[0]
+
+        assert article.subject_id == old_subject
+
+        article.Edite(title, content, subject)
+
+        assert article.title == title
+        assert article.content == content
+        assert article.subject_id == subject
+        # if subject not in Article.SUBJECTS the value won't be changed
+        unsaved_subject = Subject(name='k')
+        article.Edite(subject=unsaved_subject)
+        assert article.subject_id == subject
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_searchObjects(self, create_articles, create_supjects, create_user):
+        user1 = create_user[1]
```

Comment:
Please use for loops

Reply:
can't do it in create_user , im passing diffrent arguments,
instead of do it in two lines it gonna be 5-7 lines

Reply:
less organized

Reply:
@Anas-mhameed
Any comments here?

Reply:
I didn't understand exactly what you mean,
if you mean to loop in create_user that is kind defficult , i have deffirent args for each object , and looping there will just complicate the code

Reply:
Maybe you should have more tests 
test_search_for_exist_article
test_search_for_not_exist_article

Reply:
done